### PR TITLE
Lab2

### PR DESCRIPTION
**Pull request by Khalil19-99**

### CODE REVIEW

**Comment by mcflydesigner** [comment]
Great job! Approved.

**Comment by nailyav** [comment]
Looks good

**Comment by mcflydesigner** [comment]
Both Docker images work from docker hub correctly!

File: app_python/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM python:3.9-slim
+
+RUN mkdir -p /app
```
Comment (e2xen) [inline app_python/Dockerfile:3]: You could avoid doing this `RUN` command (and adding one more container layer) by  `COPY requirements.txt /app/requirements.txt`
`WORKDIR` will still work, even if the directory itself is not created yet

File: app_python/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+WORKDIR /app
+
+RUN useradd -m runner && chown -R runner /app
+
+COPY requirements.txt requirements.txt
+RUN pip3 install --no-cache-dir -r requirements.txt
```
Comment (e2xen) [inline app_python/Dockerfile:10]: It is a good practice to add `--upgrade` when installing with `pip`

**Comment by e2xen** [comment]
Other than that, good job, approved

File: app_python/Dockerfile (hunk 2 above)
Comment (Khalil19-99) [reply app_python/Dockerfile:10]: Thanks, I'll add it

**Pull request closed without merging**